examples/ownable-two-step/tests/ownable_two_step.rs

```diff
@@ -0,0 +1,331 @@
+
+    let contract_addr = alice
+        .as_deployer()
+        .with_constructor(ctr(bob_addr))
+        .deploy()
+        .await?
+        .address()?;
+    let contract = Ownable2Step::new(contract_addr, &alice.wallet);
+
+    let err = send!(contract.transferOwnership(bob_addr))
+        .expect_err("should not transfer when not owned");
+    err.reverted_with(Ownable2Step::OwnableUnauthorizedAccount {
+        account: alice_addr,
+    });
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn accept_ownership(alice: Account, bob: Account) -> Result<()> {
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+
+    let contract_addr = alice
+        .as_deployer()
+        .with_constructor(ctr(alice_addr))
+        .deploy()
+        .await?
+        .address()?;
+    let contract = Ownable2Step::new(contract_addr, &alice.wallet);
+    receipt!(contract.transferOwnership(bob_addr))?;
+
+    // Connect as Bob and accept ownership
+    let contract = Ownable2Step::new(contract_addr, &bob.wallet);
+    let receipt = receipt!(contract.acceptOwnership())?;
+    assert!(receipt.emits(OwnershipTransferred {
+        previousOwner: alice_addr,
+        newOwner: bob_addr,
+    }));
+
+    let Ownable2Step::ownerReturn { owner } = contract.owner().call().await?;
+    assert_eq!(owner, bob_addr);
+
+    let Ownable2Step::pendingOwnerReturn { pendingOwner } =
+        contract.pendingOwner().call().await?;
+    assert_eq!(pendingOwner, Address::ZERO);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn transfer_ownership_cancel_transfer(
+    alice: Account,
+    bob: Account,
+) -> Result<()> {
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+
+    let contract_addr = alice
+        .as_deployer()
+        .with_constructor(ctr(alice_addr))
+        .deploy()
+        .await?
+        .address()?;
+    let contract = Ownable2Step::new(contract_addr, &alice.wallet);
+    receipt!(contract.transferOwnership(bob_addr))?;
+
+    let receipt = receipt!(contract.transferOwnership(Address::ZERO))?;
+    assert!(receipt.emits(OwnershipTransferStarted {
+        previousOwner: alice_addr,
+        newOwner: Address::ZERO,
+    }));
+
+    let Ownable2Step::pendingOwnerReturn { pendingOwner } =
+        contract.pendingOwner().call().await?;
+    assert_eq!(pendingOwner, Address::ZERO);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn overwrite_previous_transfer_ownership(
+    alice: Account,
+    bob: Account,
+    charlie: Account,
+) -> Result<()> {
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+    let charlie_addr = charlie.address();
+
+    let contract_addr = alice
+        .as_deployer()
+        .with_constructor(ctr(alice_addr))
+        .deploy()
+        .await?
+        .address()?;
+
+    let contract = Ownable2Step::new(contract_addr, &alice.wallet);
+
+    let receipt = receipt!(contract.transferOwnership(bob_addr))?;
+    assert!(receipt.emits(OwnershipTransferred {
```

Comment:
Shouldn't this emit `OwnershipTransferStarted`?

Reply:
There is an issue with `alloy` version in our E2E library - #408